### PR TITLE
Feature marl changes merge

### PR DESCRIPTION
**Pull request by Gamenot**
Merge most of the marl changes into the main project.

### CODE REVIEW

**Review by JingfeiPeng — APPROVED** [review]

File: install_deps.sh [hunk 1]
```diff
@@ -19,23 +19,23 @@ function do_install_for_linux {
...
+    #           echo ""
+    #           read -p "WARNING. Is this OK? If you are unsure choose no. [Yn]" should_add_python_3_7
+    #           # second check to make sure they really want to
+    #           if [[ $should_add_python_3_7 =~ ^[yY\w]*$ ]]; then
+    #                 sudo apt-get install software-properties-common
+    #                 sudo add-apt-repository ppa:deadsnakes/ppa
+    #                 sudo apt-get install python3.7 python3.7-dev python3.7-tk python3.7-venv
+    #           fi
+    #      fi
+    # fi
```
Comment (Adaickalavan) [inline install_deps.sh:38]: Why were these lines commented out?

**Review by Adaickalavan — APPROVED** [review]

File: log/1608353019/episode_0/agent_AGENT-3.csv [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+Num_Collision,0
```
Comment (JingfeiPeng) [inline log/1608353019/episode_0/agent_AGENT-3.csv:4]: Just noticed there are some log files

File: log/1608353019/episode_0/agent_AGENT-2.csv [hunk 3]
```diff
@@ -0,0 +1,4 @@
+,0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16
+Speed,14.538060811122325,13.969758237646026,14.12055729952184,13.637937198262652,13.71695040459194,13.005271562131938,12.947920022459297,12.980917859261657,12.521986235989818,11.963820296861998,12.02981909735082,12.18851532213145,12.385374318390443,12.715496578231681,12.399002571521834,12.612734568923914,12.24910285611828
+GDistance,68.33075403630716,66.90839848188024,65.50509066963144,64.12115485741785,62.75415064137216,61.41856781387623,60.12751087364158,58.8313624631735,57.5567232412094,56.332479109937985,55.13314798097625,53.92205377595414,52.6949841112438,51.4500571582032,50.1952577784434,48.94662005069172,47.77842252964409
+Num_Collision,1
```
Comment (JingfeiPeng) [inline log/1608353019/episode_0/agent_AGENT-2.csv:4]: log file

File: install_deps.sh (hunk 1 above)
Comment (Gamenot) [reply install_deps.sh:38]: It looks like @KornbergFresnel may have had an issue with it? I will undo this before merging.

File: log/1608353019/episode_0/agent_AGENT-3.csv (hunk 2 above)
Comment (Gamenot) [reply log/1608353019/episode_0/agent_AGENT-3.csv:4]: Good catch.

**Pull request merged**